src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,99 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final double LOAD_FACTOR = 0.75;
+    private static final int DEFAULT_CAPACITY = 16;
+    private int threshold;
+    private int capacity;
+    private int size;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        capacity = DEFAULT_CAPACITY;
+        threshold = (int) (capacity * LOAD_FACTOR);
+        table = (Node<K, V>[]) new Node[capacity];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size >= threshold) {
+            table = resize();
+        }
+        setNode(new Node<>(key, value, null));
     }
 
     @Override
     public V getValue(K key) {
+        if (size == 0) {
+            return null;
+        }
+        Node<K, V> currentNode = table[getHash(key) % capacity];
+        while (currentNode != null) {
+            if (currentNode.key == key
+                    || (currentNode.key != null && currentNode.key.equals(key))) {
+                return currentNode.value;
+            }
+            currentNode = currentNode.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void setNode(Node<K, V> newNode) {
+        int indexOfBucket = getHash(newNode.key) % capacity;
+        if (table[indexOfBucket] == null) {
+            table[indexOfBucket] = newNode;
+        } else {
```

Comment:
You can use here return statement and delete the 'else' statement

Reply:
done